### PR TITLE
Implement stream metadata operations.

### PR DESCRIPTION
**Pull request by YoEight**

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -490,69 +494,397 @@ impl StreamMetadataBuilder {
...
     pub fn builder() -> StreamMetadataBuilder {
         StreamMetadataBuilder::new()
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Acl {
+    User,
+    System,
+    Stream(StreamAcl),
```
Comment (YoEight) [inline src/types.rs:628]: @oskardudycz I'd like your input on how we could name those predefined ACLs, `User` (`$userStreamAcl` internally) and `System` (`$systemStreamAcl` internally)

**Review by arwinneil — APPROVED** [review]

File: tests/integration.rs [hunk 2]
```diff
@@ -92,6 +93,53 @@ async fn test_read_stream_events(client: &Client) -> Result<(), Box<dyn Error>>
...
+async fn test_metadata(client: &Client) -> Result<(), Box<dyn Error>> {
+    let stream_id = fresh_stream_id("metadata");
+    let events = generate_events("metadata-test".to_string(), 5);
+
+    let _ = client
+        .append_to_stream(stream_id.as_str(), &Default::default(), events)
+        .await?;
+
+    let expected = StreamMetadataBuilder::new()
+        .max_age(2)
```
Comment (arwinneil) [inline tests/integration.rs:105]: [Failing CI test](https://github.com/EventStore/EventStoreDB-Client-Rust/pull/62/checks?check_run_id=2628545113#step:3:443) : `max_age()` takes Duration

```suggestion
        .max_age(Duration::from_secs(2))
```

Comment (YoEight) [reply tests/integration.rs]: Done.

**Review by arwinneil — APPROVED** [review]

File: src/types.rs (hunk 1 above)
Comment (YoEight) [reply src/types.rs:675]: Done

**Review by oskardudycz — APPROVED** [review]

**Pull request merged**